src/legacy/ui/public/styles/_legacy/components/_sidebar.scss

```diff
@@ -112,8 +112,13 @@
       align-self: center;
     }
   }
+}
 
-  .index-pattern-selection:not(.euiComboBox) {
-    padding: $euiSizeS 0;
-  }
+.indexPattern__container {
+  display: flex;
+  align-items: center;
+}
```

Comment:
To the eye this looks good, but I do see that Lens adds a couple of more styles to their equivalent selector `.lnsInnerIndexPatternDataPanel__header`. For consistency, we should probably match this more closely by adding the missing rules.

```
height: $euiSize * 3;
margin-top: -$euiSizeS;
```

See here: https://github.com/elastic/kibana/blob/ac0f44e6a50235b8673d1541fec21cfd8ffc548c/x-pack/legacy/plugins/lens/public/indexpattern_plugin/_datapanel.scss#L7-L12